client/flutter/lib/pages/onboarding/notifications_page.dart

```diff
@@ -25,6 +31,19 @@ class _NotificationsPageState extends State<NotificationsPage> {
   }
 
   void _allowNotifications() async {
+    // iOS needs a call for permissions but Android doesn't.
+    // If the user opts-out on Android we just don't register the device token.
+    await _firebaseMessaging.requestNotificationPermissions(
+        const IosNotificationSettings(
+            sound: true, badge: true, alert: true, provisional: false));
+
+    _firebaseMessaging.onIosSettingsRegistered
```

Comment:
I'm still a bit confused by the async flow that's happening here. This line looks like it's registering a callback to respond to the `onIosSettingsRegistered` event, but the event registration doesn't happen until after the `requestNotificationPermissions` async function returns. What exactly is the `IosSettingsRegistered` event?